src/test/java/tqs/example/impostor/repository/AdminRepositoryTest.java

```diff
@@ -0,0 +1,52 @@
+package tqs.example.impostor.repository;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@DataJpaTest
+public class AdminRepositoryTest {
+
+    @Autowired
+
+    private AdminRepository adminRepository;
+    private Admin testAdmin1;
+
+    @BeforeEach
+    void setUp(){
+        testAdmin1 = new Admin("Stonoga", "PanJestZerem");
+        adminRepository.saveAndFlush(testAdmin1);
+    }
+
+    @Test
+    void findById_WhenGoodId(){
+        Optional<Admin> found = adminRepository.findById(testAdmin1.getId());
+        if(found.isEmpty()) Assertions.fail();
+        assertThat(found.get().getId()).isEqualTo(testAdmin1.getId());
+    }
+
+    @Test
+    void findById_WhenBadId(){
+        Optional<Admin> found = adminRepository.findById(2137420L);
+        assertThat(found).isEmpty();
+    }
+
+    @Test
+    void findByUserName_WhenGoodUserName(){
+        Optional<Admin> found = adminRepository.findByUserName(testAdmin1.getUserName());
+        if(found.isEmpty()) Assertions.fail();
+        assertThat(found.get().getUserName()).isEqualTo(testAdmin1.getUserName());
+    }
+
+    @Test
+    void findByUserName_WhenBadUserName(){
```

Comment:
QA has to finally specify the naming conventions, because it starting to get messy. For now I propose that the test method naming convention should be as follows: "given[here the parameter that the function is given]_when[function name]_then[the correct return type/value]" all in pascal case.